sdk/attestation/azure-security-attestation/src/private/attestation_deserializers_private.cpp

```diff
@@ -0,0 +1,297 @@
+  using namespace Azure::Security::Attestation::Models::_detail;
+  using namespace Azure::Security::Attestation;
+  using namespace Azure::Core::Json::_internal;
+  /***************************************
+   * A quick note on the naming convention for the Serialize/Deserialize classes.
+   *
+   * Serialization classes are named "XxxxSerializer". They contain one or two static methods
+   * named "Serialize" and "Deserialize".
+   *
+   * The Serialize method takes an instance of an "Xxxx" object and returns a std::string which
+   * represents the Xxxx object serialized into JSON.
+   *
+   * The Deserialize method takes an instance of a json object and returns an instance of the Xxxx
+   * type.
+   *
+   */
+
+  AttestationOpenIdMetadata OpenIdMetadataSerializer::Deserialize(
+      std::unique_ptr<Azure::Core::Http::RawResponse>& response)
+  {
+    Models::AttestationOpenIdMetadata returnValue;
+    auto parsedBody = Azure::Core::Json::_internal::json::parse(response->GetBody());
+
+    JsonOptional::SetIfExists(returnValue.Issuer, parsedBody, "issuer");
+    JsonOptional::SetIfExists(returnValue.JsonWebKeySetUrl, parsedBody, "jwks_uri");
+    JsonOptional::SetIfExists(returnValue.SupportedClaims, parsedBody, "claims_supported");
+    JsonOptional::SetIfExists(
+        returnValue.SupportedTokenSigningAlgorithms,
+        parsedBody,
+        "id_token_signing_alg_values_supported");
+    JsonOptional::SetIfExists(
+        returnValue.SupportedResponseTypes, parsedBody, "response_types_supported");
+    return returnValue;
+  }
+
+  std::string AttestSgxEnclaveRequestSerializer::Serialize(AttestSgxEnclaveRequest const& request)
+  {
+    Azure::Core::Json::_internal::json serializedRequest;
+    serializedRequest["quote"] = Azure::Core::_internal::Base64Url::Base64UrlEncode(request.Quote);
+
+    JsonOptional::SetFromNullable<AttestationData, std::map<std::string, std::string>>(
+        request.RunTimeData,
+        serializedRequest,
+        "runtimeData",
+        [](AttestationData const& value) -> std::map<std::string, std::string> {
+          return {
+              {"data", Azure::Core::_internal::Base64Url::Base64UrlEncode(value.Data)},
+              {"dataType", value.DataType.ToString()}};
+        });
+
+    JsonOptional::SetFromNullable<AttestationData, std::map<std::string, std::string>>(
+        request.InitTimeData,
+        serializedRequest,
+        "inittimeData",
+        [](AttestationData const& value) -> std::map<std::string, std::string> {
+          return {
+              {"data", Azure::Core::_internal::Base64Url::Base64UrlEncode(value.Data)},
+              {"dataType", value.DataType.ToString()}};
+        });
+
+    JsonOptional::SetFromNullable(request.Nonce, serializedRequest, "nonce");
+    JsonOptional::SetFromNullable(
+        request.DraftPolicyForAttestation, serializedRequest, "draftPolicyForAttestation");
+    return serializedRequest.dump();
+  }
```

Comment:
would be worth it to define the function and re-use instead of defining it twice:


```suggestion
  std::string AttestSgxEnclaveRequestSerializer::Serialize(AttestSgxEnclaveRequest const& request)
  {
    Azure::Core::Json::_internal::json serializedRequest;
    serializedRequest["quote"] = Azure::Core::_internal::Base64Url::Base64UrlEncode(request.Quote);

    auto decorator = [](AttestationData const& value) -> std::map<std::string, std::string> {
          return {
              {"data", Azure::Core::_internal::Base64Url::Base64UrlEncode(value.Data)},
              {"dataType", value.DataType.ToString()}};
        };

    JsonOptional::SetFromNullable<AttestationData, std::map<std::string, std::string>>(
        request.RunTimeData,
        serializedRequest,
        "runtimeData",
        decorator
        );

    JsonOptional::SetFromNullable<AttestationData, std::map<std::string, std::string>>(
        request.InitTimeData,
        serializedRequest,
        "inittimeData",
        decorator
        );

    JsonOptional::SetFromNullable(request.Nonce, serializedRequest, "nonce");
    JsonOptional::SetFromNullable(
        request.DraftPolicyForAttestation, serializedRequest, "draftPolicyForAttestation");
    return serializedRequest.dump();
  }
```